torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -478,29 +484,51 @@ void ProcessGroupNCCL::ncclCommWatchdogInternal() {
         if (checkForNCCLErrors(ncclComms)) {
           LOG(INFO) << "Received NCCL errors for communicators in the cache";
 
-          if (blockingWait_) {
-            LOG(INFO) << "Aborting communicators that received errors";
-            // We should not abort the communicators if we are performing a
-            // non-blocking wait(). The reason for this is that if we abort the
-            // nccl communicator, wait() might not throw exceptions and
-            // subsequent operations might run on garbage results.
-            // The current model is that when we call wait(), subsequent
-            // operations only run after this work is done or we hang forever
-            // waiting for the operation to complete.
-            for (const auto& ncclComm : ncclComms) {
-              ncclComm->ncclCommAbort();
-              // Note that we don't remove the aborted communicators from the
-              // cache. The reason is that if we do remove the communicator
-              // from the cache, it is possible that a new collective operation
-              // calls `ncclCommInitRank` to create a new communicator whereas
-              // other ranks might have failed/timed out and didn't enter
-              // `ncclCommInitRank`. As a result, when there is a failure on
-              // a communicator the application receives an exception and its
-              // their responsibility to destroy the process group and recreate
-              // it to recover from errors.
-              abortedCommIds.emplace(
-                  buildNcclUniqueIdStr(ncclComm->getNcclId()));
-            }
+          LOG(INFO) << "Aborting communicators that received errors";
+          // We should not abort the communicators if we are performing a
+          // non-blocking wait(). The reason for this is that if we abort the
+          // nccl communicator, wait() might not throw exceptions and
+          // subsequent operations might run on garbage results.
+          // The current model is that when we call wait(), subsequent
+          // operations only run after this work is done or we hang forever
+          // waiting for the operation to complete.
```

Comment:
It seems like we violate the contract mentioned here if we remove `blockingWait_` here?

Reply:
I guess users will get the same behavior when `blockingWait_` is true? Verified on HPC trainer, blocking_wait just works with lower QPS

Reply:
If `NCCL_BLOCKING_WAIT` is false and `NCCL_ASYNC_ERROR_HANDLING` is false, we would still end up aborting communicators here that might cause consistency issues where other ops after the aborted collective might run on corrupted data.

Reply:
Good catch - thanks. We should probably guard this code block with `if (blockingWait_ || asyncErrorHandling_)` to handle this case.